### PR TITLE
Fix documented definition of ``UGate``

### LINKED ISSUE #7197: Wrong expression in UGate docs
### Information

- **Qiskit Terra version**: 0.18.13
- **Python version**: All
- **Operating system**: All

### What is the current behavior?

The documentation for [`UGate`](https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/circuit/library/standard_gates/u.py) states that:

```
U(\theta, \phi, \lambda) =
    RZ(\phi - \pi/2) RX(\pi/2) RZ(\pi - \theta) RX(\pi/2) RZ(\lambda - \pi/2)
```

### Steps to reproduce the problem

Problem is in the code documentation of [`UGate`](https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/circuit/library/standard_gates/u.py).

### What is the expected behavior?

The expression above does not correspond to the matrix form given in the documentation of [`UGate`](https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/circuit/library/standard_gates/u.py), nor to the RX and RY expressions in terms of U. Since the latter are all in line with the behaviour of Qiskit's ideal simulator, the expression itself is incorrect. The correct expression is instead:

```
U(\theta, \phi, \lambda) =
    RZ(\phi) RX(\pi/2) RZ(\pi - \theta) RX(\pi/2) RZ(\lambda - \pi)
```

### Suggested solutions

Fix the expression in the documentation. For convenience of the reader, one could also add the following simplified expression, from which the matrix, RX and RY expressions are more readily derived:

```
U(\theta, \phi, \lambda) =
    RZ(\phi + \pi/2) RX(\theta) RZ(\lambda - \pi/2)
```

### PR DESCRIPTION
**Pull request by sg495**
### Summary

Fixes #7197.

### CODE REVIEW

**Comment by jakelishman** [comment]
I left some comments on the underlying issue (#7197) - let's talk there first, then we can update the PR if needed.

**Review by jakelishman — COMMENTED** [review]
Sorry I was absent for so long on this - it slipped through the gaps in the run up to Terra 0.19.  It'd be good to be consistent with `U3Gate` because they implement the same operation. I think the form there might be a shade neater, and it's easier (for me, at least) to read off the equivalence relations from that; the simplified form there is clearly just `RZ(\phi) @ RY(\theta) @ RZ(\lambda)`, and the other equivalence follows trivially from that.

**Comment by sg495** [comment]
No problem. What can I do to help on my end?

**Comment by jakelishman** [comment]
If you agree / are happy to do it, I just meant that the first expression (line 31) should agree with the [definition in the `U3Gate` documentation](https://qiskit.org/documentation/stubs/qiskit.circuit.library.U3Gate.html), and keep the nice simplified form you've given.

**Comment by HuangJunye** [comment]
@sg495 would you still be interested in finishing the PR?

File: qiskit/circuit/library/standard_gates/u.py [hunk 1]
```diff
@@ -28,7 +28,13 @@ class UGate(Gate):
 
     .. math::
         U(\theta, \phi, \lambda) =
-            RZ(\phi - \pi/2) RX(\pi/2) RZ(\pi - \theta) RX(\pi/2) RZ(\lambda - \pi/2)
+            RZ(\phi) RX(\pi/2) RZ(\pi - \theta) RX(\pi/2) RZ(\lambda - \pi)
```
Comment (Cryoris) [inline qiskit/circuit/library/standard_gates/u.py:31]:
```suggestion
            RZ(\phi) RX(-\pi/2) RZ(\theta) RX(\pi/2) RZ(\lambda)
```

**Pull request merged**